Lib/asyncio/selector_events.py

```diff
@@ -864,6 +864,8 @@ def __del__(self, _warn=warnings.warn):
         if self._sock is not None:
             _warn(f"unclosed transport {self!r}", ResourceWarning, source=self)
             self._sock.close()
+            if self._server is not None:
+                self._server._detach(self)
```

Comment:
Interesting. I take it this was a bug in the original `__del__` code that you discovered during testing?

Reply:
Indeed. If `wait_closed()` was waiting for a transport that was gc:d, then it would never finish.